changelogs/current.yaml

```diff
@@ -134,6 +134,9 @@ bug_fixes:
 - area: tcp_proxy
   change: |
     When tunneling TCP over HTTP, mark the upstream connection as done reading when upstream trailers are read. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.finish_reading_on_decode_trailers`` to false.
+- area: router
+  change: |
+    fixed a bug that custom tags of the route metadata type are not set for upstream spans.
```

Comment:
I would frame this as a new feature, I don't think this ever worked. I also would refer to it "upstream access logs" instead of spans

Reply:
Do you mean dynamic/cluster metadata of access logs? I guess this PR is not related to access logs? I set the `route_` in `stream_info_`, and I think it only enables us to add custom tags of route metadata to the upstream spans after [this change](https://github.com/envoyproxy/envoy/pull/22987/files#diff-36ebe2d9da39e88ceeccb197a9ff0b8a1dc97a5f72c8597c40d27f61a9ead00f).

Maybe you mean a *new* `ROUTE_METADATA` in access log? I found there's no `ROUTE_METADATA` in access log now. If we need to support `ROUTE_METADATA` in access log, I think we need to change substitution_formatter.cc. Then we may need a new PR?

Reply:
The main/original usage of StreamInfo is to provide information through access logs, e.g. via %ROUTE_NAME% and other things that would read from the route handle on StreamInfo, so I was thinking this was a fix for that. Sounds like its also a fix for trace spans so lets mention both?

Reply:
Got it. Done.